ww4ff/data/transform/augment.py

```diff
@@ -154,26 +160,36 @@ def augment(self, param, examples: Sequence[EmplacableExample], **kwargs):
 
 
 class DatasetMixer(AugmentModule):
-    def __init__(self, background_noise_dataset: AudioClipDataset):
-        super().__init__()
+    def __init__(self,
+                 background_noise_dataset: AudioClipDataset,
+                 do_replace: bool = False,
+                 **kwargs):
+        self.do_replace = do_replace
+        super().__init__(**kwargs)
         self.dataset = background_noise_dataset
 
     @property
     def default_params(self):
-        return AugmentationParameter([0.1, 0.2, 0.3, 0.4, 0.5], 'strength', 3),
+        return (AugmentationParameter([0.1, 0.2, 0.3, 0.4, 0.5], 'strength', 1),
+                AugmentationParameter([0], 'replace', 0, prob=0.1 if self.do_replace else 0))
 
     @torch.no_grad()
     def augment(self, param, examples: Sequence[EmplacableExample], **kwargs):
         new_examples = []
         for example in examples:
             waveform = example.audio_data
-            bg_ex = random.choice(self.dataset).audio_data.to(waveform.device)
-            b = random.randint(waveform.size(-1), bg_ex.size(-1))
+            bg_ex = self.rand.choice(self.dataset).audio_data.to(waveform.device)
+            while bg_ex.size(-1) < waveform.size(-1):
+                bg_ex = self.rand.choice(self.dataset).audio_data.to(waveform.device)
+            b = self.rand.randint(waveform.size(-1), bg_ex.size(-1))
             a = b - waveform.size(-1)
             bg_audio = bg_ex[..., a:b]
-            alpha = random.random() * param.magnitude
+            alpha = 1 if param.name == 'replace' else self.rand.random() * param.magnitude
             mixed_wf = waveform * (1 - alpha) + bg_audio * alpha
-            new_examples.append(example.emplaced_audio_data(mixed_wf))
+            ex = example.emplaced_audio_data(mixed_wf)
+            if alpha == 1:
+                ex.frame_labels = {}
```

Comment:
So if replace, it just replace existing audio sample to be a noise sample and update frame_label accordingly?, Seems like the ww label is not being updated

Reply:
The batchifier uses only `frame_labels` I believe.